Update Haskell package set to Stackage Nightly 2021-04-23 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-04-30 20:00 UTC+2**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other. 

Please note that this will be **last** merge of `haskell-updates` that I'll do for the foreseeable future. Consequently, this is going to be the last livestream about Haskell/Nix for the foreseeable future, too.

TODO
--------

- [x] Respond to the questions in https://github.com/NixOS/nixpkgs/issues/121140.

minijackson committed:
haskellPackages.pandoc-sidenote: remove broken

peti committed:
Stackage Nightly 2021-04-23

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/2fedb31fe1abdbfb2f224486b4a8b689af85024d.

magthe committed:
haskellPackages.openapi3: unbreak

magthe committed:
haskellPackages.servant-openapi3: unbreak

cdepillabout committed:
Merge pull request #120467 from magthe/haskell-updates

Unbreak openapi3 and servant-openapi3

vaibhavsagar committed:
configuration-ghc-9.0.x.nix: patch newer alex

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/470825d7ca9852419dca19e839d583053eca486f.

cdepillabout committed:
Merge pull request #120535 from vaibhavsagar/dont-patch-alex

configuration-ghc-9.0.x.nix: patch newer alex

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/7a3819ffd0724c2ad64967266a0b74353ec1816d.

sternenseemann committed:
haskellPackages.orgstat: unmark as broken

Upstream resolved the compilation issue with more recent base:
https://github.com/volhovm/orgstat/issues/15

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/ef920675c69d0a73d87ff602a3869faf0d5ae0d5.

shlevy committed:
haskellPackages.HTF: Fix testsuite and mark unbroken in configuration.

shlevy committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/ef920675c69d0a73d87ff602a3869faf0d5ae0d5.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/5ddc13836f9d266484b1b439c1a56749fdb6b0db.

cdepillabout committed:
Merge pull request #120849 from shlevy/unbreak-htf

Unbreak haskellPackages.HTF

peti committed:
Stackage Nightly 2021-04-28

peti committed:
Merge origin/master into haskell-updates.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/bbe9c771211c7a54c4e91431a29c76b2c8850d3b.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/im-retiring-from-haskell-maintenance-in-nixpkgs/12739/1

peti commented:
> https://discourse.nixos.org/t/im-retiring-from-haskell-maintenance-in-nixpkgs/12739/1

tl;dr: The upcoming livestream on Friday is going to be the last one. I'm retiring. Let those young folk maintain Haskell packages in Nix from now. :-)

peti committed:
hackage2nix: update list of broken packages

... so that there are no failing builds on Hydra.

Ping @rkrzr because icepeak is broken.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/8185884e7b585302717c6c20323b3cfaaee33967.

peti committed:
git-annex: update sha256 hash for the new version
